src/viadot/sources/business_core.py

```diff
@@ -0,0 +1,177 @@
+        headers = {"Content-Type": "application/x-www-form-urlencoded"}
+        response = handle_api_response(
+            url=url,
+            headers=headers,
+            method="GET",
+            data=payload,
+            verify=self.verify,
+        )
+        token = json.loads(response.text).get("access_token")
+        self.token = token
+        return token
+
+    def clean_filters_dict(self) -> dict:
+        """Function for replacing 'None' with '&' in a dictionary.
+
+            Needed for payload in 'x-www-form-urlencoded' from.
+
+        Returns:
+            Dict: Dictionary with filters prepared for further use.
+        """
+        return {
+            key: ("&" if val is None else val) for key, val in self.filters_dict.items()
+        }
+
+    def get_data(self) -> dict:
+        """Function for obtaining data in dictionary format from Business Core API.
+
+        Returns:
+            Dict: Dictionary with data downloaded from Business Core API.
+        """
+        filters = self.clean_filters_dict()
+
+        payload = (
+            "BucketCount="
+            + str(filters.get("BucketCount"))
+            + "BucketNo="
+            + str(filters.get("BucketNo"))
+            + "FromDate="
+            + str(filters.get("FromDate"))
+            + "ToDate"
+            + str(filters.get("ToDate"))
+        )
+        headers = {
+            "Content-Type": "application/x-www-form-urlencoded",
+            "Authorization": "Bearer " + self.generate_token(),
+        }
+        self.logger.info("Downloading the data...")
+        response = handle_api_response(
+            url=self.url,
+            headers=headers,
+            method="GET",
+            data=payload,
+            verify=self.verify,
+        )
+        self.logger.info("Data was downloaded successfully.")
+        return json.loads(response.text)
+
+    def to_df(self, if_empty: Literal["warn", "fail", "skip"] = "skip") -> pd.DataFrame:
+        """Function for transforming data from dictionary to pd.DataFrame.
+
+        Args:
+            if_empty (Literal["warn", "fail", "skip"], optional):
+            What to do if output DataFrame is empty. Defaults to "skip".
+
+        Returns:
+            pd.DataFrame: DataFrame with data downloaded from Business Core API view.
+
+        Raises:
+            APIError: When selected API view is not available.
+        """
+        view = self.url.split("/")[-1]
+
+        if view not in [
+            "GetCustomerData",
+            "GetItemMaster",
+            "GetPendingSalesOrderData",
+            "GetSalesInvoiceData",
+            "GetSalesReturnDetailData",
+            "GetSalesOrderData",
+            "GetSalesQuotationData",
+        ]:
+            error_message = f"View {view} currently not available."
+            raise APIError(error_message)
+
+        data = self.get_data().get("MasterDataList")
```

Comment:
`get_data()`/`to_records()` should already return the data we're interested in, so this filtering should be done there